test/test_solvers.jl

```diff
@@ -178,8 +179,109 @@ function poisson_ppn_recover_sine_cosine_solution(FT, Nx, Ny, Nz, Lx, Ly, Lz, mx
     isapprox(ϕ, Ψ.(xC, yC, zC); rtol=5e-2)
 end
 
-@testset "Poisson solvers" begin
-    println("Testing Poisson solvers...")
+function can_solve_single_tridiagonal_system(arch, N)
+    ArrayType = array_type(arch)
+
+    a = rand(N-1) |> ArrayType
+    b = 3 .+ rand(N) |> ArrayType  # +3 to ensure diagonal dominance.
+    c = rand(N-1) |> ArrayType
+    f = rand(N) |> ArrayType
+
+    M = Tridiagonal(a, b, c)
+    ϕ_correct = M \ f  # This solve probably invokes scalar CuArray operations on the GPU!
```

Comment:
You could solve this on the CPU for testing purposes? It doesn't matter what machine is used to generate the correct result.

Reply:
True. They could end up calling different libraries/algorithms but the answer shouldn't be different for well-conditioned matrices.

I'll perform the backslash on the CPU to avoid scalar operations.